R/sampler-hmc.R

```diff
@@ -145,7 +145,7 @@ hmc_transform <- function(domain) {
   }
 
   ## Save finite bounds for later:
-  a <- lower[is_bounded | is_semi_infinite]
+  a <- lower[is_bounded]
```

Comment:
Why have we reverted this?

Reply:
It was causing a warning in the subsetting in the tests, so it wasn't equivalent in the end. We can revisit it later, but I think it was only a theoretical suggested change originally?